documentation/inventory_files.md

```diff
@@ -175,6 +175,81 @@ target or group of targets:
   Get-BoltInventory -Targets <TARGET LIST> -Detail
   ```
 
+## Loading inventory
+
+Bolt only loads inventory for commands that accept targeting options (`query`,
+`rerun`, and `targets`). For example, Bolt loads inventory when it runs a task
+but does not load inventory when it installs modules.
+
+When Bolt loads inventory, it loads the entirety of the inventory, not just
+the targets or groups provided to the targeting option. Because of this, it's
+important to understand how the inventory is resolved, as loading the entire
+inventory can impact performance and result in different target configuration
+than you might expect.
+
+For example, the following inventory includes two groups: `ssh` and `winrm`.
+Both groups define a target named `server`, which is configured to use a
+different transport and authentication credentials dependening on which group is
+referenced:
+
+```yaml
+# ~/.puppetlabs/bolt/inventory.yaml
+groups:
+  - name: ssh
+    targets:
+      - uri: server.example.com
+        name: server
+        config:
+          transport: ssh
+          ssh:
+            user: root
+            password: SSHP@ssword!
+  - name: winrm
+    targets:
+      - uri: server.example.com
+        name: server
+        config:
+          transport: winrm
+          winrm:
+            user: admin
+            password: WinRMP@ssword!
+```
+
+If you run a command on the `ssh` group, the `server` target uses the SSH
+transport and configuration. However, if you run a command on the `winrm`
+group, the `server` target still uses the SSH transport and configuration,
+not the WinRM transport and configuration. You can see this by showing
+the inventory detail for the `winrm` group:
+
+```shell
+$ bolt inventory show --targets winrm --detail
+
+server
+  name: server
+  uri: server.example.com
+  config:
+    transport: ssh
+    ssh:
+      password: SSHP@ssword!
+      user: root
+  groups:
+  - ssh
+  - winrm
+  - all
+
+...
+```
+
+Even though the command only specifies a single group, the entirety of the
+inventory file is loaded. Because the `server` target is defined in both the
```

Comment:
```suggestion
Even though the command only specifies a single group, Bolt loads the entire
inventory file. Because the `server` target is defined in both the
```